src/components/SidePanels/TabPanels/utils.ts

```diff
@@ -0,0 +1,189 @@
+import { FC, ReactNode } from 'react';
+import { DEFAULT_PANEL_HEIGHT, DEFAULT_PANEL_MAX_HEIGHT, DEFAULT_PANEL_WIDTH } from '../constants';
+import { Details } from '../DetailsPanel/DetailsPanel';
+import { OutlinerComponent } from '../OutlinerPanel/OutlinerPanel';
+import { PanelProps } from '../PanelBase';
+import { DroppableSide, PanelBBox, PanelView } from './types';
+
+export const lastCallTime: { [key: string]: number } | undefined = {};
+export const timeouts: { [key: string]: ReturnType<typeof setTimeout> | null } = {};
+  
+export const throttleAndPop = (callback: () => any, callback2: () => any, name:string, delay = 100): any => {
+  const now = Date.now();
+  const timeSinceLastCall = now - (lastCallTime[name] || 0);
+  
+  if (timeSinceLastCall >= delay) {
+    lastCallTime[name] = now;
+    callback();
+    clearTimeout(timeouts[name]!);
+    timeouts[name] = setTimeout(() => {
+      lastCallTime[name] = 0;
+      timeouts[name] = null;
+      callback2();
+    }, delay * 2);
+  }
+};
```

Comment:
Can this be replaced with just using lodash debounce? Not sure what this does exactly as it appears to not be throttling but debouncing the calls.